service/history/workflow/mutable_state_impl.go

```diff
@@ -2801,6 +2798,54 @@ func (ms *MutableStateImpl) ApplyWorkflowExecutionStartedEvent(
 	return nil
 }
 
+func (ms *MutableStateImpl) AddWorkflowExecutionPausedEvent(
```

Comment:
do we need to first check is workflow is already in paused state or not? or that will be done in the api handler?